modules/migration-cleaning-up-failed-migration.adoc

```diff
@@ -0,0 +1,132 @@
+
+
+// Module included in the following assemblies:
+//
+// * migrating_from_ocp_3_to_4/troubleshooting-3-4.adoc
+// * migration-toolkit-for-containers/troubleshooting-mtc
+
+
+:_module-type: PROCEDURE
+
+[id="cleanup-of-failed-migration_{context}"]
+= Cleaning up a failed migration
+
+.Prerequisites
+* You must be logged in to the {product-title} cluster as a user with the `cluster-admin` role.
+* You must have the OpenShift CLI installed.
+
+
+.Procedure
+
+
+. Deleting resources
++
+If a migration fails during Stage or Copy, some resources are retained to allow debugging and need to be cleaned up.
++
+The following resources must be deleted:
+
+* rsync/stunnel pods, both in the source and target namespaces
+* Secrets and configmap, both in the source and target namespaces
+* Route and service only in the target namespaces
+
+
+
+
+. Unquiescing applications
++
+If your application was quiesced during migration, you must unquiesce it by scaling it back to its initial replica count. The original replica count is indicated in a label that MTC adds to the `deployment` resource when a source application is quiesced during migration:
+
++
+----
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+ annotations:
+    deployment.kubernetes.io/revision: "1"
+    migration.openshift.io/preQuiesceReplicas: "1"
+----
+Unquiesce the application in either of the following two ways:
+
+** Manually edit the deployment primitive (`Deployment`, `DeploymentConfig`, etc.) and set the `spec.replicas` field back to its original, non-zero value:
++
+----
+$ oc edit deployment <deployment_name>
+----
+
+** Scale your deployment with the `oc scale` command:
++
+----
+$ oc scale deployment <deployment_name> --replicas=<desired_replicas>
+----
+
+
+. Deleting the MTC Operator and cluster-scoped resources
+
++
+CAUTION: This procedure entirely removes Velero. Perform it only if there are no more Velero instances employed on the cluster.
+
+.. Delete the Migration Controller and its resources:
++
+[source,terminal]
+----
+$ oc delete migrationcontroller <resource_name>
+----
++
+Wait for the MTC Operator to finish deleting the resources.
+
+.. Uninstall the MTC Operator:
+
+   ** OpenShift 4: Uninstall the Operator in the [web console](https://docs.openshift.com/container-platform/4.7/operators/olm-deleting-operators-from-cluster.html) LINK ERROR or by running the following command:
```

Comment:
1. This would be an xref, not a link.
2. No xrefs allowed in modules.
3. No links to docs.openshift.com. That domain is EOL soon.
4. What is "LINK ERROR"?

Reply:
Link error - this link does not work.

Reply:
Please review the difference between link and xref.

Reply:
It might help if you did not use a markdown link format in an asciidoc document. That's why it does not work. Markdown and Asciidoc are different formats. Also, please review the difference between xref and link.